File: leetcode-04/solutions/cpp/0485.cpp
class Solution {
 public:
  int findMaxConsecutiveOnes(vector<int>& nums) {
    int ans = 0;
    int sum = 0;

    for (int num : nums) {
      sum += num;
      if (num == 0)
        sum = 0;
      else
        ans = max(ans, sum);
    }

    return ans;
  }
};

File: leetcode-04/solutions/python3/0485.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        ans = 0
        sum = 0

        for num in nums:
            sum += num
            if num == 0:
                sum = 0
            else:
                ans = max(ans, sum)

        return ans
